aws-amplify/amplify-android
# feat(datastore): Add recoverability improvements

*Issue #, if available:*

- Fixes: #1637
- Fixes: #1794
- Fixes: #1865 
- Fixes: #2077 

*Description of changes:*
- Call errorHandler on mutation errors
- ~Log network responses~
- Don't die on non-json responses
- Remove unused constant
- ~Log WebSocket incoming/outgoing messages~
- ~Remove~ Deprecate `isSyncRetryEnabled` from `AWSDataStorePlugin`
-  ~Remove~ Deprecate `doSyncRetry` from `DataStoreConfiguration`
- Log ConflictHandler calls
- Adjust non-retryable errors in `MutationProcessor` and `SyncProcessor`
- Align RetryHandler logic with other platforms
- ~Remove optional retry of errors in sync~ Log a warning when disabling sync retries

*How did you test these changes?*
- Wrote additional unit tests
- Manual testing with emulator
- Simulation of different network conditions via a MITM (e.g. 5XX, 4XX, timeouts, connection resets, etc)


*Documentation update required?*
  - aws-amplify/docs#4986

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
